### PR TITLE
feature (ref DPLAN 11857): use DuplicateInterIdException to send right mess…

### PR DESCRIPTION
**Pull request by jalalifardemos**
### Ticket: https://demoseurope.youtrack.cloud/issue/DPLAN-11857/Excel-Import-von-Stellungnahme-geht-nicht-mit-Beispieldatei-mit-Sauke-Aramaki

Description: If excel import has a same Eingangsnummer (InternId), Error message is not clear.

Delete the checkbox if it doesn't apply/isn't necessary.

- [ ] Tests updated/created
- [ ] Update documentation
- [ ] Link all relevant tickets
- [x] Move the tickets on the board accordingly
- [ ] Data-Cy attributes added/updated ([conventions](https://dplan-documentation.demos-europe.eu/development/guidelines-conventions/coding-styleguides/twig_html.html#guideline-for-naming-cypress-hooks))

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php [hunk 1]
```diff
@@ -2509,11 +2509,18 @@ public function importStatementsFromXls(
             throw new DemosException(self::STATEMENT_IMPORT_ENCOUNTERED_ERRORS);
         } catch (Exception $e) {
             $this->logger->error(self::STATEMENT_IMPORT_ENCOUNTERED_ERRORS, ['exception' => $e]);
-            $this->getMessageBag()->add(
-                'error',
-                'statements.import.error.document.unexpected',
-                ['doc' => $fileInfo->getFileName()]
-            );
+            if (DuplicateInternIdException::class) {
```
Comment (sierrazamudiodemos) [inline demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php:2512]: if (DuplicateInternIdException::class === get_class($e)) I guess this is missing, the rest looks good thanks for using this exception

File: demosplan/DemosPlanCoreBundle/Repository/StatementRepository.php [hunk 2]
```diff
@@ -192,6 +193,9 @@ public function addObject($statement): Statement
             return $statement;
         } catch (Exception $e) {
             $this->getLogger()->warning('Add StatementObject failed Message: ', [$e]);
+            if (str_contains($e->getMessage(), 'internId_procedure')) {
```
Comment (sierrazamudiodemos) [inline demosplan/DemosPlanCoreBundle/Repository/StatementRepository.php:196]: I was wondering if the internID is empty, is it throwing this exception or another one?

**Review by sierrazamudiodemos — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Repository/StatementRepository.php (hunk 2 above)
Comment (jalalifardemos) [reply demosplan/DemosPlanCoreBundle/Repository/StatementRepository.php:196]: InternId can be null

Comment (sierrazamudiodemos) [reply demosplan/DemosPlanCoreBundle/Repository/StatementRepository.php:196]: catch (UniqueConstraintViolationException $e) {
            if (str_contains($e->getMessage(), 'internId_procedure')) {
                throw DuplicateInternIdException::create('Eingangsnummer', $statement->getProcedureId());
            }
            throw $e;
        }

**Review by sierrazamudiodemos — APPROVED** [review]

**Pull request merged**